Calibration/IsolatedParticles/plugins/StudyCaloResponse.cc

```diff
@@ -419,16 +435,20 @@ void StudyCaloResponse::analyze(edm::Event const& iEvent, edm::EventSetup const&
 
     edm::Handle<reco::VertexCollection> recVtxs;
     iEvent.getByToken(tok_recVtx_, recVtxs);
-    int ntrk(0), ngoodPV(0), nPV(-1);
-    int nvtxs = (int)(recVtxs->size());
-    for (int ind = 0; ind < nvtxs; ind++) {
-      if (!((*recVtxs)[ind].isFake()) && (*recVtxs)[ind].ndof() > 4)
-        ngoodPV++;
-    }
-    for (int i = 0; i < nPVBin_; ++i) {
-      if (ngoodPV >= pvBin_[i] && ngoodPV < pvBin_[i + 1]) {
-        nPV = i;
-        break;
+    int ntrk(0), ngoodPV(0), nPV(-1), nvtxs(0);
+    if (!recVtxs.isValid()) {
```

Comment:
Shouldnt you just not run this when there is no valid vertex? Give an exception or something?

Reply:
Doing that